Banganza - by Ida and Tilde

Idahel opened the pull request:
https://banganza.netlify.app/
https://project-final-pfy9.onrender.com/

admin login:
Username: tilde&ida-final-project
Password: letsmeetatilcaffe
--> https://banganza.netlify.app/admin

tildetilde committed:
Added dynamic background

tildetilde committed:
Deleted shaders

tildetilde committed:
Merge pull request #1 from tildetilde/file-structure

Deleted shaders

tildetilde committed:
Create TODO.md

tildetilde committed:
Merge branch 'main' of https://github.com/tildetilde/project-final

Idahel committed:
basic backend set up

Idahel committed:
Merge pull request #2 from tildetilde/backend-set-up

basic backend set up

tildetilde committed:
Design setup

tildetilde committed:
Merge branch 'main' of https://github.com/tildetilde/project-final

tildetilde committed:
tailwind fix

tildetilde committed:
tailwind config

tildetilde committed:
tailwind config

tildetilde committed:
config taiwlind

tildetilde committed:
design test

tildetilde committed:
tailwind setup

tildetilde committed:
added ui components

tildetilde committed:
tailwind config

tildetilde committed:
Design system

tildetilde committed:
Merge pull request #3 from tildetilde/creating-ui-parts

Creating UI parts and design system

tildetilde committed:
Building pages

tildetilde committed:
Added routes

tildetilde committed:
STyled playing cards

tildetilde committed:
Merge pull request #4 from tildetilde/styling-TimeLineCard

Styled playing cards

tildetilde committed:
Login file setup

tildetilde committed:
Merge pull request #5 from tildetilde/login-modal

Login file setup

Idahel committed:
backend before refactoring

Idahel committed:
Merge pull request #6 from tildetilde/connect-spotify-api

backend before refactoring

tildetilde committed:
Login modal up

tildetilde committed:
Merge pull request #7 from tildetilde/login-modal

Login modal up

tildetilde committed:
Added orientation guard

tildetilde committed:
Merge pull request #8 from tildetilde/game-mode

Added orientation guard

Idahel committed:
login backend + frontend

tildetilde committed:
Game mode update

tildetilde committed:
Merge pull request #9 from tildetilde/game-mode

Game mode update

Idahel committed:
Add token status endpoint and enhanced token management

Idahel committed:
Merge branch 'main' into connect-login

Idahel committed:
Merge pull request #10 from tildetilde/connect-login

Connect login

tildetilde committed:
improving game mode

tildetilde committed:
Merge pull request #11 from tildetilde/game-mode

improving game mode

Idahel committed:
frontend

Idahel committed:
Merge pull request #12 from tildetilde/connect-login

frontend

Idahel committed:
trying to fix sign in

Idahel committed:
Merge pull request #13 from tildetilde/fix-login

trying to fix sign in

Idahel committed:
trying to make authflow correct

Idahel committed:
Merge pull request #14 from tildetilde/auth-fix

trying to make authflow correct

Idahel committed:
spotify auth fixed

Idahel committed:
Merge pull request #15 from tildetilde/spotify-auth

spotify auth fixed

tildetilde committed:
Update package.json

tildetilde committed:
Mockup data and improved timeline

tildetilde committed:
Merge pull request #16 from tildetilde/improving-game-mode

Mockup data and improved timeline

tildetilde committed:
Adjusting currentCard being draggable

tildetilde committed:
Merge pull request #17 from tildetilde/improving-game-mode

Adjusting currentCard being draggable

tildetilde committed:
improved drag and drop for timeline

tildetilde committed:
Merge pull request #18 from tildetilde/improving-game-mode

improved drag and drop for timeline

tildetilde committed:
Improved start card logic

tildetilde committed:
Merge pull request #19 from tildetilde/improving-game-mode

Improved start card logic

tildetilde committed:
Option to change position in Timeline

tildetilde committed:
Merge pull request #20 from tildetilde/improving-game-mode

Function to change position of currentCard in Timeline

tildetilde committed:
Visual Feedback for right and wrong

tildetilde committed:
Merge pull request #21 from tildetilde/feedback-for-right-and-wrong

Visual Feedback for right and wrong

tildetilde committed:
Logic for points

tildetilde committed:
Merge pull request #22 from tildetilde/points-logic-for-team

Logic for points

Idahel committed:
cleaned up index.tserror handling

Idahel committed:
Merge pull request #23 from tildetilde/check-auth

cleaned up index.tserror handling

tildetilde committed:
New loading page

tildetilde committed:
Merge pull request #24 from tildetilde/new-loading-page

New loading page

tildetilde committed:
Erasing mockup sign in

tildetilde committed:
Merge pull request #25 from tildetilde/reset-of-mockup

Erasing mockup sign in

tildetilde committed:
Updated

tildetilde committed:
Merge pull request #26 from tildetilde/reset-of-mockup

Updated environement

tildetilde committed:
New intro and home

tildetilde committed:
Merge pull request #27 from tildetilde/creating-landing-page

New intro and home

tildetilde committed:
Hard reset to mack data

tildetilde committed:
Merge remote-tracking branch 'origin/main' into restore-loading-page

tildetilde committed:
Merge pull request #28 from tildetilde/restore-loading-page

Hard reset to mack data

tildetilde committed:
Using own data in timeline

tildetilde committed:
Merge pull request #29 from tildetilde/new-game-mode

Using own data in timeline

tildetilde committed:
Using own data with no mockup data

tildetilde committed:
Merge pull request #30 from tildetilde/new-game-mode

Using own data with no mockup data

tildetilde committed:
Improved design

tildetilde committed:
Merge pull request #31 from tildetilde/new-game-mode

Improved design

tildetilde committed:
Improved design

tildetilde committed:
Merge pull request #32 from tildetilde/design-of-game-mode

Improved design

Idahel committed:
changed backend from using spotify api and auth to our own quiz data

Idahel committed:
Merge pull request #33 from tildetilde/backend-clean-up

changed backend from using spotify api and auth to our own quiz data

tildetilde committed:
Approved corner layout of gamemode

tildetilde committed:
Merge pull request #34 from tildetilde/design-of-game-mode

Approved corner layout of gamemode

tildetilde committed:
Updated design of timeline

tildetilde committed:
Merge pull request #35 from tildetilde/design-of-game-mode

Updated design of timeline

tildetilde committed:
Update GameMode.tsx

tildetilde committed:
Merge pull request #36 from tildetilde/improving-timeline

Altered position of timeline

Idahel committed:
added question and isViseble to api response

Idahel committed:
Merge pull request #37 from tildetilde/connect-frontend-backend

added question and isViseble to api response

tildetilde committed:
Erased old API leftovers (artist, TrackCard etc)

tildetilde committed:
Merge pull request #38 from tildetilde/deleting-all-from-sptoify-api

Erased old API leftovers (artist, TrackCard etc)

Idahel committed:
clean up

Idahel committed:
clean up

Idahel committed:
Merge pull request #39 from tildetilde/clean-up-backend

Clean up backend

tildetilde committed:
Added coastline

tildetilde committed:
Merge pull request #40 from tildetilde/tilde-backend-and-frontend

Added coastline category and item

Idahel committed:
frontend and backend connected

Idahel committed:
Merge pull request #41 from tildetilde/set-up-connection-frontend-backend

frontend and backend connected

Idahel committed:
updated timeline card, current card and gameMode view

Idahel committed:
Merge pull request #42 from tildetilde/fix-category-cards

updated timeline card, current card and gameMode view

Idahel committed:
updated data

Idahel committed:
Merge pull request #43 from tildetilde/max-10-card-win

updated data

Idahel committed:
added random logic for item card render using shuffle

Idahel committed:
Merge pull request #44 from tildetilde/game-logic

added random logic for item card render using shuffle

tildetilde committed:
Home have categories

tildetilde committed:
Merge pull request #45 from tildetilde/improve-home-page

moved category selector from game mode to home

tildetilde committed:
Added placeholder game settings

tildetilde committed:
Merge pull request #46 from tildetilde/improve-home-page

Added placeholder game settings

Idahel committed:
removed unused import

Idahel committed:
Merge pull request #47 from tildetilde/game-board-width

removed unused import

Idahel committed:
admin page to delete, edit and create new data to quiz

Idahel committed:
Merge pull request #48 from tildetilde/admin-page

admin page to delete, edit and create new data to quiz

tildetilde committed:
game settings up and running

tildetilde committed:
Merge pull request #49 from tildetilde/tilde-game-settings

game settings up and running

tildetilde committed:
Update GameMode.tsx

tildetilde committed:
Merge pull request #50 from tildetilde/game-mode-fix

Update GameMode.tsx

tildetilde committed:
Improving Timer logic

tildetilde committed:
Merge pull request #51 from tildetilde/imporivng-game-settings

Improving Timer logic

tildetilde committed:
Created Winner function

tildetilde committed:
Merge pull request #52 from tildetilde/tilde-winner-function

Created Winner function

tildetilde committed:
Current team showing in game mode

tildetilde committed:
Merge pull request #53 from tildetilde/tilde-gamesettings-polish

Current team showing in game mode

tildetilde committed:
Winner feedback

tildetilde committed:
Merge pull request #54 from tildetilde/tilde-gamesettings-polish

Winner feedback

Idahel committed:
added game reset and confirmation pop up for homepage link on game mode board

Idahel committed:
Merge pull request #55 from tildetilde/reset-game

added game reset and confirmation pop up for homepage link on game mo…

tildetilde committed:
Winner feedback + options after game is over

tildetilde committed:
Merge pull request #56 from tildetilde/tilde-gamesettings-polish

Winner feedback + options after game is over

tildetilde committed:
Limiting intro using session storage

tildetilde committed:
Merge pull request #57 from tildetilde/tilde-set-intro-as-done

Limiting intro using session storage

tildetilde committed:
changed team names in game settings

tildetilde committed:
Merge pull request #58 from tildetilde/timeline-look

changed team names in game settings

tildetilde committed:
less wide timeline

tildetilde committed:
Merge pull request #59 from tildetilde/tilde-timeline-width

less wide timeline

Idahel committed:
added deployed backend URL

Idahel committed:
Merge pull request #60 from tildetilde/use-deployed-backend

added deployed backend URL

tildetilde committed:
Update GameBoard.tsx

Idahel committed:
added netlify redirects and toml

Idahel committed:
Merge pull request #61 from tildetilde/check-admin

added netlify redirects and toml

Idahel committed:
changed netlify toml

Idahel committed:
Merge pull request #62 from tildetilde/netlify

changed netlify toml

tildetilde committed:
Implemented spinner

tildetilde committed:
Merge pull request #63 from tildetilde/tilde-loading-spinner

Implemented spinner

tildetilde committed:
new intro

tildetilde committed:
Merge pull request #64 from tildetilde/tilde-itnro

new intro

tildetilde committed:
improved intro

tildetilde committed:
Merge pull request #65 from tildetilde/tilde-itnro

improved intro

tildetilde committed:
improved intro and home screen

tildetilde committed:
Merge pull request #66 from tildetilde/tilde-itnro

improved intro and home screen

Idahel committed:
new width to game settings container

Idahel committed:
Merge pull request #67 from tildetilde/game-settings-update

new width to game settings container

tildetilde committed:
Update banganzalogo.svg

tildetilde committed:
Merge pull request #68 from tildetilde/tilde-itnro

Update banganzalogo.svg

Idahel committed:
adjusted card sizes

Idahel committed:
Merge pull request #69 from tildetilde/timeline-cards-resize

adjusted card sizes

Idahel committed:
removed dots from the header, changed scaling on current card and timeline cards, moved instructions to drag card

Idahel committed:
Merge pull request #70 from tildetilde/remove-dot-pattern

removed dots from the header, changed scaling on current card and tim…

tildetilde committed:
New category look

tildetilde committed:
Merge pull request #71 from tildetilde/tilde-new-category-cards

New category look

Idahel committed:
resized cards: timeline, current card. Changed size on heading, removed heading from game settings

Idahel committed:
Merge pull request #72 from tildetilde/restore-cards

resized cards: timeline, current card. Changed size on heading, remov…

Idahel committed:
game reset when choosing new category

Idahel committed:
Merge pull request #73 from tildetilde/reset-game-when-finished

game reset when choosing new category

tildetilde committed:
new icons and question names

tildetilde committed:
Merge pull request #74 from tildetilde/tilde-category-card

new icons and question names

Idahel committed:
changed gap

Idahel committed:
Merge pull request #75 from tildetilde/fix-spacing-drag-and-drop

changed gap

tildetilde committed:
improved home

tildetilde committed:
Merge pull request #76 from tildetilde/tilde-category-card

improved home

Idahel committed:
icons - kebab-case to camelCase

Idahel committed:
Merge pull request #77 from tildetilde/error-fix

icons - kebab-case to camelCase

Idahel committed:
added better game feedback

Idahel committed:
Merge pull request #78 from tildetilde/game-feedback

added better game feedback

Idahel committed:
changed card font to mono

Idahel committed:
Merge pull request #79 from tildetilde/game-design

changed card font to mono

tildetilde committed:
improved home

tildetilde committed:
Merge pull request #80 from tildetilde/tilde-improving-home

improved home

tildetilde committed:
improved layout

tildetilde committed:
Merge pull request #81 from tildetilde/tilde-design-improvements

improved layout

tildetilde committed:
connected font

tildetilde committed:
Merge pull request #82 from tildetilde/tilde-design-improvements

connected font

tildetilde committed:
Improved feedback system and card logic

tildetilde committed:
Merge pull request #83 from tildetilde/tilde-feedback-and-font-review

Improved feedback system and card logic

Idahel committed:
inverted colors for game mode

Idahel committed:
Merge branch 'main' into darkmode

Idahel committed:
Merge pull request #84 from tildetilde/darkmode

inverted colors for game mode

Idahel committed:
fixed missed closing tag

Idahel committed:
small design changes

Idahel committed:
Merge pull request #85 from tildetilde/remove-from-game-settings

small design changes

Idahel committed:
changed drag and drop to work better on mobile

Idahel committed:
Merge pull request #86 from tildetilde/drag-and-drop-mobile

changed drag and drop to work better on mobile

Idahel committed:
mobile view

Idahel committed:
Merge pull request #87 from tildetilde/game-board-mobile

mobile view

Idahel committed:
adjustments to fit mobile

Idahel committed:
addded spacing for lg team names

Idahel committed:
Merge pull request #88 from tildetilde/mobile-adjustments

adjustments to fit mobile

Idahel committed:
Merge pull request #89 from tildetilde/mobile-adjustments

addded spacing for lg team names

Idahel committed:
changed border color

Idahel committed:
Merge pull request #90 from tildetilde/mobile-adjustments

changed border color

tildetilde committed:
Accessibility 100

tildetilde committed:
Merge pull request #91 from tildetilde/tilde-checking-accessibility

Accessibility 100

tildetilde committed:
added more cateogires

tildetilde committed:
Merge pull request #92 from tildetilde/more-categories

added more cateogires

tildetilde committed:
Deleting redundant code

tildetilde committed:
Merge pull request #93 from tildetilde/tilde-cleaning-up-frontend

Deleting redundant code

tildetilde committed:
Cleaning up

tildetilde committed:
Merge pull request #94 from tildetilde/tilde-cleaning

Cleaning up

tildetilde committed:
cleaning up

tildetilde committed:
Merge pull request #95 from tildetilde/tilde-cleaning

cleaning up

tildetilde committed:
align eslint deps (ts-eslint v7) + route prune via frontend

tildetilde committed:
Merge pull request #96 from tildetilde/tilde-production-mode

align eslint deps (ts-eslint v7) + route prune via frontend

tildetilde committed:
Update README.md

tildetilde committed:
Update README.md

Idahel committed:
production mode

Idahel committed:
Merge pull request #97 from tildetilde/production-mode

production mode

Idahel committed:
Move @types to dependencies for production build

Idahel committed:
Merge pull request #98 from tildetilde/production-mode

Move @types to dependencies for production build

Idahel committed:
cleaned up backend, unnecessary complex

Idahel committed:
Merge pull request #99 from tildetilde/production-mode

cleaned up backend, unnecessary complex

Idahel committed:
removed duplicate shuffle logic (backend and frontend), changed frontend shuffle logic to Fisher-Yates (indusry standard and more reliable), fixed items/category cards and filering options for smaller screens, added break-word and hypens for game cards as the text somtimes overflows

Idahel committed:
Merge pull request #100 from tildetilde/production-mode

removed duplicate shuffle logic (backend and frontend), changed front…

Idahel committed:
removed unused interfaces from types.ts, removed import of logger.ts and replaced costum logger with consol log in seedDb and create admin

Idahel committed:
Merge pull request #101 from tildetilde/development

removed unused interfaces from types.ts, removed import of logger.ts …

JennieDalgren reviewed (approved):
Great job with this project!

You are approved with VG

Idahel committed:
removed round from game mode, added a card count for current player

Idahel committed:
Merge pull request #102 from tildetilde/development

removed round from game mode, added a card count for current player, display correct answer when fail

Idahel committed:
changes styling for cardcount

Idahel committed:
Merge pull request #103 from tildetilde/center-counter-text

changes styling for cardcount

Idahel committed:
made game mode top content absolute for better experience on mobile

Idahel committed:
Merge pull request #104 from tildetilde/gamemode-sticy

made game mode top content absolute for better experience on mobile

tildetilde committed:
Update game data and enhance category API handling

Corrected release year for Tetris and replaced God of War with The Sims 2 in games.items.json. Improved TimeLineCard to respect unit visibility. Extended API service to include more category fields (description, unitVisible, sort, version). Updated asset references in index.html.

tildetilde committed:
Merge pull request #105 from tildetilde/tilde-checkout-project

Update game data and enhance category API handling

tildetilde committed:
Add game info modal to Home page

Introduces an accessible modal dialog with game instructions, accessible via a new info button on the Home page. The modal can be closed with the Escape key or by clicking outside, and includes detailed setup and gameplay instructions for users.

tildetilde committed:
Merge pull request #106 from tildetilde/tilde-checkout-project

Add game info modal to Home page

tildetilde committed:
Add in-game info modal to GameMode page

Introduces an accessible info modal explaining game rules and setup, accessible via a new 'Info' button in both pre-game and in-game phases. The modal can be closed with the Escape key or a close button, improving user guidance and onboarding.
